cdap-app-fabric/src/main/java/io/cdap/cdap/gateway/handlers/ProgramLifecycleHttpHandler.java

```diff
@@ -2136,4 +2141,25 @@ private ProgramId batchProgramToProgramId(String namespace, BatchProgram batchPr
                            batchProgram.getProgramId());
     }
   }
+
+  /**
+   * Convert BatchProgram to ProgramIds with all app versions
+   */
+  private List<ProgramId> batchProgramToProgramIds(String namespace, BatchProgram batchProgram) {
+    try {
+      Collection<ApplicationSpecification> allAppVersions = store.getAllAppVersions(
+        new ApplicationId(namespace, batchProgram.getAppId()));
+      return allAppVersions.stream().map(applicationSpecification -> {
+                                           ApplicationId applicationId = new ApplicationId(
+                                             namespace, applicationSpecification.getName(),
```

Comment:
can applicationSpecification be null? In that case need to handle applicationSpecification.getName()